src/QueryPipeline/QueryPipeline.cpp

```diff
@@ -273,7 +274,28 @@ static void initRowsBeforeLimit(IOutputFormat * output_format)
         output_format->setRowsBeforeLimitCounter(rows_before_limit_at_least);
     }
 }
+static void initRowsBeforeAggregation(std::shared_ptr<Processors> processors, IOutputFormat * output_format)
+{
+    bool has_aggregation = false;
 
+    if (!processors->empty())
+    {
+        RowsBeforeAggregationCounterPtr rows_before_aggregation_at_least = std::make_shared<RowsBeforeStepCounter>();
+        for (auto processor : *processors)
+        {
+            if (auto transform = std::dynamic_pointer_cast<AggregatingTransform>(processor))
```

Comment:
Also take into account `AggregatingInOrderTransform`.

Reply:
Thank you very much for your reminder. I will take it into account.